### PR TITLE
Add collectionAssertion for the new infix API

### PR DESCRIPTION
**Pull request by ratkayandras**
----
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/master/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1338,9 +1338,9 @@ expect(Paths.get("/usr/bin/noprogram")).exists()
 ```
 ↑ <sub>[Example](https://github.com/robstoll/atrium/tree/master/samples/readme-examples/src/main/kotlin/readme/examples/ReadmeSpec.kt#L326)</sub> ↓ <sub>Output</sub>
 ```text
-expect: /usr/bin/noprogram        (sun.nio.fs.UnixPath <1234789>)
+expect: \usr\bin\noprogram        (sun.nio.fs.WindowsPath <1234789>)
```
Comment (robstoll) [inline README.md]: please revert the changes in this file

File: apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/CollectionFeatureAssertionsSpec.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package ch.tutteli.atrium.api.infix.en_GB
+
+import ch.tutteli.atrium.creating.Expect
+import ch.tutteli.atrium.specs.fun1
+import ch.tutteli.atrium.specs.name
+import ch.tutteli.atrium.specs.notImplemented
+import ch.tutteli.atrium.specs.property
+
+class CollectionFeatureAssertionsSpec : ch.tutteli.atrium.specs.integration.CollectionFeatureAssertionsSpec(
+    property<Collection<String>, Int>(Expect<Collection<String>>::size).name to Expect<Collection<String>>::size,
```
Comment (robstoll) [inline apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/CollectionFeatureAssertionsSpec.kt]: since you don't use a companion function you can use directly
```suggestion
    property<Collection<String>, Int>(Expect<Collection<String>>::size),
```

File: apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/collectionAssertions.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
+package ch.tutteli.atrium.api.infix.en_GB
+
+import ch.tutteli.atrium.creating.Expect
+import ch.tutteli.atrium.domain.builders.ExpectImpl
+
+object Empty
```
Comment (robstoll) [inline apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/collectionAssertions.kt]: please copy the parameterObjects from api-cc-infix-en_GB-common to the new api and remove this object

File: README.md (hunk 1 above)
Comment (ratkayandras) [reply README.md]: Okay I reverted that, will be up in the next push

File: apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/CollectionFeatureAssertionsSpec.kt (hunk 2 above)
Comment (ratkayandras) [reply apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/CollectionFeatureAssertionsSpec.kt]: I changed  that, will be up in the next push

File: apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/collectionAssertions.kt (hunk 3 above)
Comment (ratkayandras) [reply apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/collectionAssertions.kt]: I copied the keywords file, but it has a lot of imports from `ch.tutteli.atrium.api.cc.infix.en_GB` like:

```
import ch.tutteli.atrium.api.cc.infix.en_GB.and
import ch.tutteli.atrium.api.cc.infix.en_GB.grouped
import ch.tutteli.atrium.api.cc.infix.en_GB.ignoring
import ch.tutteli.atrium.api.cc.infix.en_GB.inAny
import ch.tutteli.atrium.api.cc.infix.en_GB.to
import ch.tutteli.atrium.api.cc.infix.en_GB.within
```
where 'infix' is unresolved reference.

These are from multiple files and I don't know if you want hem in the new infix api's common package

Comment (robstoll) [reply apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/collectionAssertions.kt]: Ah... my fault, not parameterObjects but keywords/keywords 
We will use parameterObjects eventually as well but just copy keywords for now

Comment (ratkayandras) [reply apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/collectionAssertions.kt]: Sorry my modifications to keywords.kt hasn't been pushed, I am gonna fix it in the next

**Comment by robstoll** [comment]
@ratkayandras thanks for the contribution :+1: 

Fancy another one? 
https://github.com/robstoll/atrium/issues?q=is%3Aissue+is%3Aopen+label%3Ahacktoberfest+no%3Aassignee

**Comment by matevarga** [comment]
Andras did this as a part of a recruitment process -- we (Patients Know Best) ask people to contribute to an OSS project (instead of giving them pointless made-up coding exercises). We found this issue on https://up-for-grabs.net

**Comment by robstoll** [comment]
Cool idea 🙂👍
